op-acceptance-tests/tests/interop/loadtest/max_execute_test.go

```diff
@@ -0,0 +1,142 @@
+package loadtest
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-devstack/devtest"
+	"github.com/ethereum-optimism/optimism/op-service/txintent"
+	suptypes "github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+type messageSource interface {
+	Get() *suptypes.Message
+}
+
+type messageSink interface {
+	Append(*suptypes.Message)
+}
+
+type messagePool struct {
+	mu       sync.Mutex // mu protects both rng's rand.Source and messages.
+	rng      *rand.Rand
+	messages []*suptypes.Message
+}
+
+var _ messageSource = (*messagePool)(nil)
+var _ messageSink = (*messagePool)(nil)
+
+// newMessagePool creates a messagePool, using message as the first message in the pool.
+func newMessagePool(message *suptypes.Message) *messagePool {
+	return &messagePool{
+		rng:      rand.New(rand.NewSource(1234)),
+		messages: []*suptypes.Message{message},
+	}
+}
+
+// Get pseudorandomly selects a message from the pool.
+func (p *messagePool) Get() *suptypes.Message {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	index := p.rng.Intn(len(p.messages))
+	return p.messages[index]
+}
+
+// Append adds msg to the pool.
+func (p *messagePool) Append(msg *suptypes.Message) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	p.messages = append(p.messages, msg)
+}
+
+// MaxExecutor is a spammer that executes messages from a source and sends those executing messages
+// as initiating messages to a sink.
+type MaxExecutor struct {
+	l2     *L2
+	source messageSource
+	sink   messageSink
+}
+
+var _ Spammer = (*MaxExecutor)(nil)
+
+func NewMaxExecutor(source messageSource, sink messageSink, dest *L2) *MaxExecutor {
+	return &MaxExecutor{
+		l2:     dest,
+		source: source,
+		sink:   sink,
+	}
+}
+
+func (e *MaxExecutor) Spam(t devtest.T) error {
+	// Get an initiating message from the source and execute it.
+	initMsg := e.source.Get()
+	start := time.Now()
+	tx, err := e.l2.Include(t, planExecMsg(t, initMsg, e.l2.BlockTime, e.l2.EL.Escape().EthClient()))
+	if err != nil {
+		return fmt.Errorf("include exec msg: %w", err)
+	}
+	messageLatency.WithLabelValues(e.l2.Config.ChainID.String(), "exec").Observe(time.Since(start).Seconds())
+
+	// All executing messages are initiating messages. Send the executing message to the sink as
+	// an initiating message.
+	initMsg, err = initMsgFromReceipt(t, e.l2, tx.Receipt)
+	if err != nil {
+		return err
+	}
+	e.sink.Append(initMsg)
+
+	return nil
+}
+
+// TestMaxExecutingMessagesBurst runs the MaxExecutor spammer on a Burst schedule on both chains.
+// The executing messages emitted by one spammer become initiating messages for the other spammer.
```

Comment:
Is there any reason to do this except for the elegance of never needing to emit new messages?

Reply:
It makes the supervisor do more work. It is easier for the supervisor to handle pure init msgs. If every init message is also an exec message, then it makes the supervisor perform access list checks in addition to the usual indexing.